rero_ils/modules/items/api.py

```diff
@@ -192,12 +198,30 @@ def update(self, data, dbcommit=False, reindex=False):
         :returns: The updated item record.
         """
         data = trim_barcode_for_record(data=data)
+        if not data.get('barcode'):
```

Comment:
We have the same code in the replace function. A `set_barcode` function could be usefull.